daemon/config/config_test.go

```diff
@@ -153,6 +153,32 @@ func TestDaemonConfigurationMergeConflictsWithInnerStructs(t *testing.T) {
 	}
 }
 
+// Test for #40711
+func TestDaemonConfigurationMergeFlagsWitchConfigFile(t *testing.T) {
+	f, err := ioutil.TempFile("", "docker-config-")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	configFile := f.Name()
+	f.Write([]byte(`{}`))
+	f.Close()
+
+	var conf = Config{}
+	flags := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	flags.Var(&conf.NetworkConfig.DefaultAddressPools, "default-address-pool", "")
+	flags.Set("default-address-pool", "base=10.123.0.0/16,size=24")
+
+	config, err := MergeDaemonConfigurations(&conf, flags, configFile)
+
+	if err != nil {
+		t.Fatalf("merge error %v", err)
+	}
```

Comment:
Given that we're already importing gotest.tools, this could be;

```suggestion
	assert.NilError(t, err)
```